src/kitsController.ts

```diff
@@ -46,17 +48,56 @@ export class KitsController {
   public static isScanningForKits() { return this.checkingHaveKits; }
 
   folderKits: Kit[] = [];
+  additionalKits: Kit[] = [];
 
   private constructor(readonly cmakeTools: CMakeTools, private readonly _kitsWatcher: chokidar.FSWatcher) { }
 
+  static async expandAdditionalKitFiles(cmakeTools: CMakeTools): Promise<string[]> {
+    const additionalKitFiles: string[] = cmakeTools.workspaceContext.config.additionalKits;
+    const expandedAdditionalKitFiles: string[] = [];
+
+    const opts: expand.ExpansionOptions = {
+      vars: {
+        buildKit: cmakeTools.activeKit?.name || "",
+        buildType: await cmakeTools.currentBuildType() || "",
+        buildKitVendor: "",
+        buildKitTriple: "",
+        buildKitVersion: "",
+        buildKitHostOs: "",
+        buildKitTargetOs: "",
+        buildKitTargetArch: "",
+        buildKitVersionMajor: "",
+        buildKitVersionMinor: "",
+        generator: "",
+        userHome: paths.userHome,
+        workspaceFolder: cmakeTools.workspaceContext.folder.uri.fsPath,
+        workspaceFolderBasename: path.basename(cmakeTools.workspaceContext.folder.uri.fsPath),
+        workspaceHash: "",
+        workspaceRoot: cmakeTools.workspaceContext.folder.uri.fsPath,
+        workspaceRootFolderName: path.basename(cmakeTools.workspaceContext.folder.uri.fsPath),
+      }
+    };
+
+    await Promise.resolve(additionalKitFiles.forEach(async kitFile => {
```

Comment:
We talked about using Promise.all here instead.  See `scanForVSKits` for an example.